src/sentry/api/endpoints/team_projects.py

```diff
@@ -27,7 +32,13 @@
 
 class ProjectPostSerializer(serializers.Serializer):
     name = serializers.CharField(max_length=50, required=True)
-    slug = serializers.RegexField(r"^[a-z0-9_\-]+$", max_length=50, required=False, allow_null=True)
+    slug = serializers.RegexField(
```

Comment:
One option would be to add flag to all these Serializers but the other option is to add flag to DEFAULT_SLUG_PATTERN, return the old pattern if not flagged. Error message can remain the same for both.